pkg/engine/armresources_test.go

```diff
@@ -1191,3 +1191,662 @@ func TestGenerateARMResourceWithVMASAgents(t *testing.T) {
+	var cs api.ContainerService
+
+	json.Unmarshal([]byte(apiModelStr), &cs)
+
+	armResources := GenerateARMResources(&cs)
+
+	tg, _ := InitializeTemplateGenerator(Context{})
+
+	expectedCustomDataStr := getCustomDataFromJSON(tg.GetKubernetesLinuxNodeCustomDataJSONObject(&cs, cs.Properties.AgentPoolProfiles[0]))
+
+	var dataDisks []compute.VirtualMachineScaleSetDataDisk
+
+	var userAssignedIDEnabled = cs.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedIDEnabled()
+
+	agentLoadBalancer := LoadBalancerARM{
+		ARMResource: ARMResource{
+			APIVersion: "[variables('apiVersionNetwork')]",
+			DependsOn: []string{
+				"[concat('Microsoft.Network/publicIPAddresses/', variables('agentPublicIPAddressName'))]",
+			},
+		},
+		LoadBalancer: network.LoadBalancer{
+			Location: to.StringPtr("[variables('location')]"),
+			Name:     to.StringPtr("[variables('agentLbName')]"),
+			LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
+				BackendAddressPools: &[]network.BackendAddressPool{
+					{
+						Name: to.StringPtr("[variables('agentLbBackendPoolName')]"),
+					},
+				},
+				FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
+					{
+						Name: to.StringPtr("[variables('agentLbIPConfigName')]"),
+						FrontendIPConfigurationPropertiesFormat: &network.FrontendIPConfigurationPropertiesFormat{
+							PublicIPAddress: &network.PublicIPAddress{
+								ID: to.StringPtr("[resourceId('Microsoft.Network/publicIpAddresses',variables('agentPublicIPAddressName'))]"),
+							},
+						},
+					},
+				},
+				OutboundRules: &[]network.OutboundRule{
+					{
+						OutboundRulePropertiesFormat: &network.OutboundRulePropertiesFormat{
+							FrontendIPConfigurations: &[]network.SubResource{
```

Comment:
I remember somewhere you need set "enableTCPReset", it's critical because the go client keep-alive problem

Reply:
Thanks for the reminder! I will add this `"enableTcpReset": true` for the outboundrule

Per this doc:
The default behavior of Load Balancer is to drop the flow silently when the outbound idle timeout has been reached. With the enableTCPReset parameter, you can enable a more predictable application behavior and control whether to send bidirectional TCP Reset (TCP RST) at the time out of outbound idle timeout.

Reply:
one more note - only set this property in k8s version later than v1.15.0, v1.14.3 and v1.13.7. This change is needed: https://github.com/kubernetes/kubernetes/issues/78011